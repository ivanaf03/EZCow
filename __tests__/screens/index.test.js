import React from "react";

import {waitFor, render, fireEvent} from "@testing-library/react-native";
import { router } from 'expo-router';

import App from "../../app/index";

jest.mock('expo-font', () => ({
    loadAsync: jest.fn(() => Promise.resolve()), 
}));

jest.mock("expo-router", () => ({
    router: {
        push: jest.fn(),
    },
}));

describe("App", () => {

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should render the correct text", () => {
        const {getByText} = render(<App />);
        waitFor (() => expect(getByText("Tu aplicación de gestión de ganado")).toBeTruthy());
        waitFor (() => expect(getByText("Iniciar sesión")).toBeTruthy());
    });

    test("should render the correct image", () => {
        const { getByTestId } = render(<App />);
        waitFor (() => expect(getByTestId("app-logo")).toBeTruthy());
    });

    it("should render the correct buttons", () => {
        const {getByText} = render(<App />);
        waitFor (() => expect(getByText("Iniciar sesión")).toBeTruthy());
        waitFor (() => expect(getByText("Registrarse")).toBeTruthy());
    });

    it("should render the correct waves", () => {
        const { getByTestId } = render(<App />);
        waitFor (() => expect(getByTestId("waves")).toBeTruthy());
    });

    it("register button should navigate to register screen", () => {
        const {getByText} = render(<App />);
        const registerButton = getByText("Registrarse");
        fireEvent.press(registerButton);
        waitFor (() => expect(router.push).toHaveBeenCalledWith("register"));
    });

    it("login button should navigate to login screen", () => {
        const {getByText} = render(<App />);
        const loginButton = getByText("Iniciar sesión");
        fireEvent.press(loginButton);
        waitFor (() => expect(router.push).toHaveBeenCalledWith("login"));
    });

});
