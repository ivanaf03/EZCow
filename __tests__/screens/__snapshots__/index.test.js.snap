// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#0E041AFF",
      },
      {
        "alignItems": "center",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "marginTop": 32,
        },
        {
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        {
          "color": "#FF8018FF",
        },
        {
          "fontSize": 72,
          "lineHeight": 72,
        },
        {
          "fontFamily": "Nunito-ExtraBold",
        },
      ]
    }
  >
    EZCow
  </Text>
  <Text
    style={
      [
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 20,
          "lineHeight": 28,
        },
        {
          "fontFamily": "Nunito-Bold",
        },
      ]
    }
  >
    Tu aplicación de gestión de ganado
  </Text>
  <Image
    source={
      {
        "testUri": "../../../assets/images/logo.png",
      }
    }
    style={
      [
        {
          "width": 384,
        },
        {
          "height": "50%",
        },
      ]
    }
    testID="app-logo"
  />
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "flexWrap": "wrap",
        },
        {
          "justifyContent": "center",
        },
        {
          "marginTop": 16,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "#75009d",
          },
          {
            "borderBottomLeftRadius": 9999,
            "borderBottomRightRadius": 9999,
            "borderTopLeftRadius": 9999,
            "borderTopRightRadius": 9999,
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "marginBottom": 8,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "fontFamily": "Nunito-Bold",
            },
            {
              "marginLeft": 8,
              "marginRight": 8,
            },
          ]
        }
      >
        Iniciar sesión
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "#75009d",
          },
          {
            "borderBottomLeftRadius": 9999,
            "borderBottomRightRadius": 9999,
            "borderTopLeftRadius": 9999,
            "borderTopRightRadius": 9999,
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "marginBottom": 8,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "fontFamily": "Nunito-Bold",
            },
            {
              "marginLeft": 8,
              "marginRight": 8,
            },
          ]
        }
      >
        Registrarse
      </Text>
    </View>
  </View>
  <Image
    source={
      {
        "testUri": "../../../assets/images/waves.png",
      }
    }
    style={
      [
        {
          "position": "absolute",
        },
        {
          "bottom": 0,
        },
        {
          "width": 384,
        },
        {
          "height": 96,
        },
      ]
    }
    testID="waves"
  />
</RNCSafeAreaView>
`;
